core/PRP/KeyedObject/plKey.h

```diff
@@ -138,10 +139,30 @@ class HSPLASMA_EXPORT plKeyData
     const plUoid& getUoid() const { return fUoid; }
 
     /** Returns a pointer to the object referenced by this key */
-    class hsKeyedObject* getObj() { return fObjPtr; }
+    template<typename _KeyedObjT = class hsKeyedObject>
+    _KeyedObjT* getObj(bool requireValid = true)
+    {
+        if constexpr (std::is_same_v<class hsKeyedObject, _KeyedObjT>) {
+            return fObjPtr;
+        } else if constexpr (std::is_base_of_v<class hsKeyedObject, _KeyedObjT>) {
+            return _KeyedObjT::Convert(fObjPtr, requireValid);
+        } else {
+            static_assert(std::is_base_of_v<class hsKeyedObject, _KeyedObjT>, "Invalid class type");
+        }
```

Comment:
I believe this can be simplified a bit to:
```suggestion
        } else {
            static_assert(std::is_base_of_v<class hsKeyedObject, _KeyedObjT>, "Invalid class type");
            return _KeyedObjT::Convert(fObjPtr, requireValid);
        }
```

Reply:
Yeah, I later simplified this to something very similar.... and neglected to push the change. 🤦‍♂️